hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/HddsDispatcher.java

```diff
@@ -410,6 +410,11 @@ ContainerCommandResponseProto createContainer(
         ContainerProtos.ContainerType.KeyValueContainer;
     createRequest.setContainerType(containerType);
 
+    if (containerRequest.hasWriteChunk()) {
```

Comment:
I think we decided to pack the index from pipeline object but not from BlockID right?

Reply:
Sorry, I am not sure about the question. What do you mean `pack from the BockID`?

I think you suggested including the replicaIndex in the block id which became a `(containerid, localid, replicaindex)` tuplet. This is the reason to have `getBlockd().getReplicaIndex()` here.

But let me know if I misunderstand something.

Reply:
I got ur point. I confused because we have not added the code to send the index from client yet, but we are trying to using here. For completion it may be good to include client side sending this idx?

Reply:
A simplified client side change is here: https://github.com/elek/ozone/commit/c86878842f29f4c5f08a16a5959c51c0a275ee3b

I think this is a well scoped and unit-tested change, client side change may need additional work. But if you would like to include this small commit, I would be happy to add it.

Reply:
It is fine for me. When need at client side we can add that.